### DISCUSSION TITLE
Why os.availableparallelism() over os.cpus().length?

### DISCUSSION THREAD

**Discussion by cnrmurphy** [nodejs/help#4743] [Q&A] [answered]
I'm just curious about what the problems are with use os.cpus().length when trying to calculate available parallelism. I see in the docs that you should use `os.availableParallelism()`, but not an explanation as to why.

**Comment by bnoordhuis** [comment, marked as answer]
https://github.com/libuv/libuv/issues/3493#issue-1150409993 - in a nutshell: os.cpus() returns the online cpus but that doesn't help you if you're pinned to a subset of them.

it's supposed to be explained in the docs but maybe it's only in the libuv docs at the moment. PR welcome if that's the case.
